### ISSUE TITLE
Allow Dynamo backends to use Inductor as fallback instead of eager mode

### ISSUE THREAD

**Issue by wenzhe-nrv** [pytorch/pytorch#94836]
### 🚀 The feature, motivation and pitch

I'm working on Dynamo oneDNN backend (being integrated at the mid-layer, like nvfuser or AITemplate) and would like to run models with mixture of Dynamo oneDNN backend + Inductor CPU backend. 

Based on oneDNN graph's programming model, we prefer to pass the full graph to oneDNN backend first, which decides its supported subgraphs, and run the rest of the graph with Inductor CPU afterward. 

This prototype PR [#90356] from @SherlockNoMad allows inductor to accepts opaque callable. This aligns with our design, where oneDNN backend will fuse subgraphs to call_module nodes and set callable for each fused node. The fused graph will be passed to Inductor CPU for compile and execute.

We'd like to check the plan of this feature and if the PR would be merged.   

### Alternatives

Another option is to pass the full graph to Inductor first, then run certain ops or patterns with oneDNN backend.

**Comment by SherlockNoMad** [comment]
cc @ngimel, @jansel

From the PR review comment, inductor would need some declaration of shape/stride for such opaque operators. Can we be specific about what is needed from external backends?

**Comment by jansel** [comment]
I believe this will be fixed once https://github.com/pytorch/pytorch/pull/90356 lands, so assigning Sherlock.  A backend could replace the subgraphs they support with opaque ops, then just pass the graph to inductor.

@SherlockNoMad for communicating shape information we basically need the moral equivalent of meta functions in primtorch.  We could also possibly automatically generate those meta functions if we have a reference implementation composed of pytorch ops.  (Which we could run fake tensors through.)  If you need help getting that over the finish line feel free to ping me or setup some time.

**Comment by ngimel** [comment]
Shouldn't backend itself provide the meta for its outputs? Backend's stride information can conceivably deviate from what running pytorch equivalent would produce.

**Comment by jansel** [comment]
Yeah, correct.   Because of that I think we need something like `inductor.opaque_op(impl, meta)`.

**Comment by tbennun** [comment]
I second the notion of an opaque op (i.e., one that survives lowering). It would also resolve the issue I posted (#95021).

**Comment by wenzhe-nrv** [comment]
@SherlockNoMad, I want to follow up with you on this. Any update or plan? Thanks.

**Comment by williamwen42** [comment]
@SherlockNoMad follow up?
